compiler/rustc_typeck/src/check/mod.rs

```diff
@@ -520,6 +521,31 @@ fn typeck_with_fallback<'tcx>(
 
             let fn_sig = fixup_opaque_types(tcx, &fn_sig);
 
+            if fn_sig.abi == abi::Abi::RustCall {
```

Comment:
I think this code should be in `check_fn`

Reply:
I moved it, I tried to put it near where I saw another ABI-related check was